server/graphql/purchase_order_line/src/mutations/insert.rs

```diff
@@ -58,26 +82,60 @@ pub fn insert_purchase_order_line(
     map_response(
         service_provider
             .purchase_order_line_service
-            .insert_purchase_order_line(&service_context, store_id, input.to_domain()),
+            .insert_purchase_order_line(&service_context, input.to_domain()),
     )
 }
 
-fn map_response(from: Result<PurchaseOrderLineRow, ServiceError>) -> Result<InsertResponse> {
-    match from {
-        Ok(purchase_order) => Ok(InsertResponse::Response(IdResponse(purchase_order.id))),
-        Err(error) => map_error(error),
-    }
+pub fn map_response(from: Result<PurchaseOrderLineRow, ServiceError>) -> Result<InsertResponse> {
+    let result = match from {
+        Ok(purchase_order_line) => {
+            InsertResponse::Response(IdResponse::from_domain(purchase_order_line))
+        }
+        Err(error) => InsertResponse::Error(InsertError {
+            error: map_error(error)?,
+        }),
+    };
+    Ok(result)
 }
 
-fn map_error(error: ServiceError) -> Result<InsertResponse> {
+pub fn map_error(error: ServiceError) -> Result<InsertErrorInterface> {
```

Comment:
While using a different end point, am exposing the same error interface for insert and insertFromCSV